examples/v1beta1/pipelineruns/pipelinerun-with-when-expressions.yaml

```diff
@@ -0,0 +1,124 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: create-readme-file
+spec:
+  resources:
+    outputs:
+      - name: workspace
+        type: git
+  steps:
+    - name: write-new-stuff
+      image: ubuntu
+      script: 'touch $(resources.outputs.workspace.path)/README.md'
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: check-file
+spec:
+  params:
+    - name: path
+  resources:
+    inputs:
+      - name: workspace
+        type: git
+  results:
+    - name: status
+      description: indicating whether the file exists
+  steps:
+    - name: check-file
+      image: alpine
+      script: |
+        if test -f $(resources.inputs.workspace.path)/$(params.path); then
+          printf exists | tee /tekton/results/status
+        else
+          printf missing | tee /tekton/results/status
+        fi
+---
+apiVersion: tekton.dev/v1alpha1
+kind: PipelineResource
+metadata:
+  name: pipeline-git
+spec:
+  type: git
+  params:
+    - name: revision
+      value: master
+    - name: url
+      value: https://github.com/tektoncd/pipeline
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: echo-file-status
+spec:
+  params:
+    - name: status
+  steps:
+    - name: echo
+      image: ubuntu
+      script: 'echo file $(params.status)'
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: guarded-pipeline
+spec:
+  resources:
+    - name: source-repo
+      type: git
+  params:
+    - name: path
+      default: "README.md"
+  tasks:
+    - name: first-create-file
+      when:
+        - input: "$(params.path)"
+          operator: in
+          values: ["README.md"]
+      taskRef:
+        name: create-readme-file
+      resources:
+        outputs:
```

Comment:
With workspaces, we should be able to avoid using `output` resources here. Clone GitHub repo into workspace and have rest of the tasks operate on that workspace.

Reply:
example was based on [the conditional pipelinerun example](https://github.com/tektoncd/pipeline/blob/master/examples/v1beta1/pipelineruns/conditional-pipelinerun.yaml), will update to use the new features, thanks 🙏